psydac/api/discretization.py

```diff
@@ -547,6 +547,20 @@ def discretize(a, *args, **kwargs):
         domain_h = args[0]
         assert isinstance(domain_h, Geometry)
         domain  = domain_h.domain
+        dim     = domain.dim
+
+        # The current implementation of the sum factorization algorithm does not support openMP parallelization
+        # It is not properly tested, whether this way of excluding openMP parallelized code from using the sum factorization algorithm works.
+        backend = kwargs.get('backend')# or None
+        assembly_backend = kwargs.get('assembly_backend')# or None
+        assembly_backend = backend or assembly_backend
+        openmp = False if assembly_backend is None else assembly_backend.get('openmp')
+
+        # Set sum_factorization to True if domain.dim==3, a is a symbolic BilinearForm and the code is not openMP parallelized ...
+        if (((isinstance(a, sym_BilinearForm)) and (dim == 3))) and not openmp:
+            # ... unless we already explicitely set sum_factorization to False! (For test cases, or in case of a newly encountered bug)
+            if not (kwargs.get('sum_factorization') == False):
+                kwargs['sum_factorization'] = True
```

Comment:
The old assembly code used a well-designed OpenMP parallelization of the vanilla assembly algorithm. The parallel scaling was great, but it started from a single-process performance which is much much slower than the new one.

So I think that there are several näive ways to use OpenMP in the sum factorization algorithm, and in most cases the resulting code will be faster than the old one. This should not take much time at all to implement, and I will be happy to help.